pkg/operator/observe_config.go

```diff
@@ -115,6 +148,23 @@ func (c ConfigObserver) sync() error {
 	return nil
 }
 
+func observeInternalRegistryHostname(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+	imageConfigClient, err := configv1client.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+	configImage, err := imageConfigClient.Images().Get("cluster", metav1.GetOptions{})
```

Comment:
ignore `NotFound` errors and just return early.  That simply means they aren't using this feature or have no customizations.